packages/autorest.python/autorest/codegen/serializers/types_serializer.py

```diff
@@ -0,0 +1,43 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from jinja2 import Environment
+from ..models import ModelType, CodeModel
+from ..models.imports import FileImport, TypingSection, ImportType
+from .import_serializer import FileImportSerializer
+
+
+class TypesSerializer:
+    def __init__(self, code_model: CodeModel, env: Environment) -> None:
+        self.code_model = code_model
+        self.env = env
+
+    def imports(self) -> FileImport:
+        file_import = FileImport()
+        if self.code_model.named_unions:
+            file_import.add_submodule_import(
+                "typing",
+                "Union",
+                ImportType.STDLIB,
+            )
+        for nu in self.code_model.named_unions:
+            if any(isinstance(t, ModelType) for t in nu.types):
```

Comment:
we should get the import here directly from the types itself, i.e. we should just have to merge the imports for each type in `nu.types`

Reply:
Yes, updated in commit https://github.com/Azure/autorest.python/pull/1733/commits/f434b5504e6b5a9cd7885054f6f0473ec9ea373f.